### PR TITLE
- add cmp/19

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **cmp/19**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/cmp/all/test_package/test_package.c [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
```
Comment (SSE4) [inline recipes/cmp/all/test_package/test_package.c:3]:
```suggestion
#include <stdbool.h>
#include <stdint.h>
```

File: recipes/cmp/all/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 3.1)
```
Comment (SSE4) [inline recipes/cmp/all/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.4)
```

File: recipes/cmp/all/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+project(cmake_wrapper)
+
+include("conanbuildinfo.cmake")
+conan_basic_setup()
```
Comment (SSE4) [inline recipes/cmp/all/CMakeLists.txt:5]:
```suggestion
conan_basic_setup()
set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```

File: recipes/cmp/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        cmake.configure(build_folder=self._build_subfolder)
+        self._cmake = cmake
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
```
Comment (SpaceIm) [inline recipes/cmp/all/conanfile.py:55]:
```suggestion
```

File: recipes/cmp/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
```
Comment (SpaceIm) [inline recipes/cmp/all/conanfile.py:61]:
```suggestion
```

File: recipes/cmp/all/test_package/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
```
Comment (SpaceIm) [inline recipes/cmp/all/test_package/CMakeLists.txt:2]:
```suggestion
project(test_package C)
```

File: recipes/cmp/all/test_package/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
```
Comment (SpaceIm) [inline recipes/cmp/all/test_package/CMakeLists.txt:5]:
```suggestion
conan_basic_setup(TARGETS)
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/cmp/all/test_package/test_package.c (hunk 1 above)
Comment (madebr) [reply recipes/cmp/all/test_package/test_package.c:3]: Are these included missing from the `"cmp.h"` header?

Comment (SSE4) [reply recipes/cmp/all/test_package/test_package.c:3]: they are not included, correct

File: recipes/cmp/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class CmpConan(ConanFile):
```
Comment (madebr) [inline recipes/cmp/all/conanfile.py:5]:
```suggestion
from conans import ConanFile, CMake, tools


required_conan_version = ">= 1.33.0"


class CmpConan(ConanFile):
```

File: recipes/cmp/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        cmake = CMake(self)
+        cmake.configure(build_folder=self._build_subfolder)
+        self._cmake = cmake
+        return self._cmake
```
Comment (madebr) [inline recipes/cmp/all/conanfile.py:48]:
```suggestion
        self._cmake = CMake(self)
        self._cmake.configure(build_folder=self._build_subfolder)
        return self._cmake
```

**Review by SpaceIm — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**